src/Meshes/transformNode.ts

```diff
@@ -1050,38 +1054,54 @@ export class TransformNode extends Node {
 
         // Billboarding (testing PG:http://www.babylonjs-playground.com/#UJEIL#13)
         if (useBillboardPath && camera) {
-            let storedTranslation = TmpVectors.Vector3[0];
-            this._worldMatrix.getTranslationToRef(storedTranslation); // Save translation
-
-            // Cancel camera rotation
-            TmpVectors.Matrix[1].copyFrom(camera.getViewMatrix());
-            TmpVectors.Matrix[1].setTranslationFromFloats(0, 0, 0);
-            TmpVectors.Matrix[1].invertToRef(TmpVectors.Matrix[0]);
-
-            if ((this.billboardMode & TransformNode.BILLBOARDMODE_ALL) !== TransformNode.BILLBOARDMODE_ALL) {
-                TmpVectors.Matrix[0].decompose(undefined, TmpVectors.Quaternion[0], undefined);
-                let eulerAngles = TmpVectors.Vector3[1];
-                TmpVectors.Quaternion[0].toEulerAnglesToRef(eulerAngles);
+            if (this.billboardMode & TransformNode.BILLBOARDMODE_USE_POSITION) {
+                this.lookAt(camera.position);
 
                 if ((this.billboardMode & TransformNode.BILLBOARDMODE_X) !== TransformNode.BILLBOARDMODE_X) {
-                    eulerAngles.x = 0;
+                    this.rotation.x = 0;

```

Comment:
The problem with that (and `this.lookAt` earlier) is that it will update values which will not be used before NEXT frame

This must happen before we evaluate `this._worldMatrix`

Reply:
Sorry I'm not seeing the right way to resolve this from your comments.  I understand the next frame issue but don't know enough about the code base to see clearly where to use the `lookAt` to effect this frame.

Looking at the `transformnode.lookAt` code its not clear that it only effects the next frame.  I'm not sure what to do from here.

Reply:
setting this.position or this.rotation is the problem as it is used before in the computeWorldMatrix function

The idea is to move your code at the beginning of the function before we compute _worldMatrix using position and rotation